### PR TITLE
Moving to scenarios

### PR DESCRIPTION
**Pull request by devonestes**
This updates the last of the three formatters to use the new `Scenario` format! 🎉

### CODE REVIEW

File: lib/benchee/formatters/html.ex [hunk 1]
```diff
@@ -114,51 +115,53 @@ defmodule Benchee.Formatters.HTML do
...
-  def format(%{statistics: statistics, run_times: run_times, system: system,
+  @spec format(Suite.t) :: %{Suite.key => String.t}
+  def format(%Suite{scenarios: scenarios, system: system,
                configuration: %{
                  formatter_options: %{html: %{file: filename}}}}) do
-    statistics
-    |> input_job_reports(run_times, system, filename)
+    scenarios
+    |> Enum.group_by(fn(scenario) -> scenario.input_name end)
+    |> Enum.map(fn(tuple) -> reports_for_input(tuple, system, filename) end)
```
Comment (PragTob) [inline lib/benchee/formatters/html.ex]: I don't think `tuple` helps much, it's `{input_name, scenarios}` right? Maybe `tagged_scenarios` or something?

File: lib/benchee/formatters/html.ex [hunk 2]
```diff
@@ -114,51 +115,53 @@ defmodule Benchee.Formatters.HTML do
...
+    sorted_stastics = scenarios
+                      |> Statistics.sort()
+                      |> Enum.map(fn(scenario) -> {scenario.job_name, scenario.run_time_statistics} end)
+                      |> Map.new
+    input_run_times = scenarios
+                      |> Enum.map(fn(scenario) -> {scenario.job_name, scenario.run_times} end)
+                      |> Map.new
     input_suite = %{
-      statistics: sorted_stats,
+      statistics: sorted_stastics,
```
Comment (PragTob) [inline lib/benchee/formatters/html.ex]: `sorted_statistics` :)

File: lib/benchee/formatters/html.ex [hunk 3]
```diff
@@ -232,9 +284,9 @@ defmodule Benchee.Formatters.HTML do
     DeviationPercent.format deviation_percent
   end
 
-  defp inputs_supplied?(input_name) do
-    input_name != Benchee.Benchmark.no_input
-  end
+  @no_input Benchee.Benchmark.no_input()
+  defp inputs_supplied?(@no_input), do: false
+  defp inputs_supplied?(_), do: true
```
Comment (PragTob) [inline lib/benchee/formatters/html.ex]: :+1:

File: lib/benchee/formatters/html.ex [hunk 4]
```diff
@@ -194,30 +197,79 @@ defmodule Benchee.Formatters.HTML do
...
       }
   """
-  def merge_job_measurements(statistics, run_times) do
-    Map.merge(statistics, run_times, fn(_key, stats, times) ->
-      %{
-        statistics: stats,
-        run_times: times
-      }
-    end)
+  def format_runtime_stats(scenarios) do
```
Comment (PragTob) [inline lib/benchee/formatters/html.ex]: I think I'd rather call it `job_measurements` or `scenario_measurements` - measurements as we have multiple values that we care about, not just the statistics but also the raw run times :) wdyt?

File: lib/benchee/formatters/html.ex (hunk 2 above)
Comment (devonestes) [inline lib/benchee/formatters/html.ex]: 😅

File: lib/benchee/formatters/html.ex (hunk 1 above)
Comment (devonestes) [inline lib/benchee/formatters/html.ex]: Or maybe `&1` :trollface: 

`tagged_scenarios` is a much better name - I'll go with that!

File: lib/benchee/formatters/html.ex (hunk 4 above)
Comment (devonestes) [inline lib/benchee/formatters/html.ex]: Looking at it again, I think `format_job_measurements` is more clear. At this point all the scenarios are for a single input, and we're grouping and formatting by job name, so I like having `job` in the function name here.

**Pull request merged**